cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -286,6 +286,145 @@ __inline__ __device__ T parse_numeric(const char* begin,
   return value * sign;
 }
 
+/**
+ * @brief Trim zeros at the beginning of raw_data
+ *
+ * @param raw_data The pointer to beginning of character string
+ * @param digit_count Total number of digits
+ */
+__device__ __inline__ void trim_zeros(char const*& raw_data, long& digit_count)
+{
+  char const* start_ptr = raw_data;
+  while (*raw_data == '0' && (raw_data < start_ptr + digit_count)) { raw_data++; }
+  digit_count -= raw_data - start_ptr;
+}
+
+/**
+ * @brief Lexicographically compare digits in input against int64 max
+ *
+ * @param raw_data The pointer to beginning of character string
+ * @return bool True if integer represented by character string is less
+ * than or equal to int64 max
+ */
+__device__ __inline__ bool lex_compare_int64max(const char* raw_data)
```

Comment:
since this returns bool, the name should be less_than_xyz, or less_equal_than_xyz. As-is, it's hard to tell what this returns at the call site.